codePost_api/helpers.py

```diff
@@ -311,7 +369,10 @@ def upload_submission(api_key, assignment, students, files, mode=DEFAULT_UPLOAD_
                 submission_id=submission_id
             )
 
-    return True
+    # Return updated submission dictionary (equivalent to what post_submission would
+    # return in the above exit paths)
+    submission = get_submission_by_id(api_key=api_key, submission_id=submission_id)
+    return submission
```

Comment:
I believe you're only setting `submission_id` in the CASE 3 branch. Perhaps the way to fix this is to capture the return value (a submission object) of `post_submission` in the other branches.

Reply:
Good catch! I think actually what needs to be done is exit the logic in CASE 1 and CASE 2, which somehow I stopped (involuntarily) doing with this update. I just pushed a commit that fixes this.